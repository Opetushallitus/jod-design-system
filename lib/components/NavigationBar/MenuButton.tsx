import { JodMenu } from '../../icons';

export interface MenuButtonProps {
  onClick: () => void;
  label: string;
}

export const MenuButton = ({ onClick, label }: MenuButtonProps) => {
  return (
    <button
      onClick={onClick}
      className="ds:flex ds:flex-col ds:md:flex-row ds:gap-2 ds:md:gap-3 ds:justify-center ds:items-center ds:select-none ds:cursor-pointer"
      data-testid="open-nav-menu"
      aria-haspopup="dialog"
    >
      <JodMenu className="ds:mx-auto" />
      <span className="ds:md:text-[14px] ds:sm:text-[12px] ds:text-[10px] ds:font-semibold">{label}</span>
    </button>
  );
};
