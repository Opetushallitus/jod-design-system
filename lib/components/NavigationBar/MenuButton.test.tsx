import { fireEvent, render, screen } from '@testing-library/react';
import { fn } from 'storybook/test';
import { describe, expect, it, vi } from 'vitest';
import { MenuButton } from './MenuButton';

describe('MenuButton', () => {
  it('renders the label text', () => {
    render(<MenuButton label="Valikko" onClick={fn()} />);
    expect(screen.getByText('Valikko')).toBeInTheDocument();
  });

  it('calls onClick when button is clicked', () => {
    const handler = vi.fn();
    render(<MenuButton label="Click me" onClick={handler} />);
    fireEvent.click(screen.getByTestId('open-nav-menu'));
    expect(handler).toHaveBeenCalledTimes(1);
  });

  it('has required accessibility and test attributes', () => {
    render(<MenuButton label="Menu" onClick={fn()} />);
    const btn = screen.getByTestId('open-nav-menu');
    expect(btn).toHaveAttribute('aria-haspopup', 'dialog');
  });
});
describe('Snapshot', () => {
  it('should render with defaults', () => {
    const { container } = render(<MenuButton label="Valikko" onClick={fn()} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
