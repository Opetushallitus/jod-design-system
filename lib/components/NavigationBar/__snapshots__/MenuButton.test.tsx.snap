// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot > should render with defaults 1`] = `
<button
  aria-haspopup="dialog"
  class="ds:flex ds:flex-col ds:md:flex-row ds:gap-2 ds:md:gap-3 ds:justify-center ds:items-center ds:select-none ds:cursor-pointer"
  data-testid="open-nav-menu"
>
  <svg
    class="ds:mx-auto"
    fill="currentColor"
    height="24"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3 18.2964V16.2964H21V18.2964H3ZM3 13.2964V11.2964H21V13.2964H3ZM3 8.29639V6.29639H21V8.29639H3Z"
      fill="currentColor"
    />
  </svg>
  <span
    class="ds:md:text-[14px] ds:sm:text-[12px] ds:text-[10px] ds:font-semibold"
  >
    Valikko
  </span>
</button>
`;
